hs/WebParsing/TimeTableParser.hs

```diff
@@ -17,15 +18,111 @@ import WebParsing.TimeConverter
+    let deptCode = getDeptCode $ parseTags body
+    let deptName = getDeptName $ parseTags body
 
     runSqlite databasePath $
         mapM_ getDeptTimetable depts
+    runSqlite databasePath $ 
+        mapM_ insertDepartment $ init $ constructAssociationList deptName deptCode
+
+-- | String left bracket to split tagText into two strings. One containing the 
+-- department name and the other the department codes. 
+bracketLeftString :: String
+bracketLeftString = "["
+
+-- | Char left bracket used to call removeUnwantedChar. Removes all char prior to 
+-- and including itself from given String. 
+bracketLeft :: Char
+bracketLeft = '['
+
+-- | Char right bracket to remove from the end of the last words in tagTexts.
+bracketRight :: Char
+bracketRight = ']'
+
+-- | List of unwanted words to remove when getting department codes.
+notDeptCodes :: [String]
+notDeptCodes = ["courses",
+                "courses]", 
+                "includes",
+                "and",
+                "-",
+                "[",
+                "]",
+                ""]
+
+-- | Parses the timetable main page, retrieves the tag text and gets the department
+-- codes in a list of lists of T.Text.
+getDeptCode :: [Tag String] -> [[T.Text]]
+getDeptCode tags = 
+    let tagsText = map fromTagText $ filter isTagText tags
+        filterDeptInfo = map (filterList bracketLeft) tagsText
+        removedEmptyStr = filter (\x -> x /= "") filterDeptInfo
+        removedChars = map (removeUnwantedChar bracketLeft) removedEmptyStr
+        listOfWords = map words removedChars
+        relevantWords = map (filter (`notElem` notDeptCodes)) listOfWords
+        deptCodesStr = map (filter (\x -> isUpperString x /= False)) relevantWords
+        deptCodesText = map (map T.pack) deptCodesStr
+    in deptCodesText
+
+-- | Parses the timetable main page, retrieves the tag text and gets the department
+-- names in a list of T.Text.
+getDeptName :: [Tag String] -> [T.Text]
+getDeptName tags = 
+    let tagsText = map fromTagText $ filter isTagText tags
+        containsBrackLeft = filter (\x -> containsChar bracketLeft x == True) tagsText
+        convertToStr = map T.pack containsBrackLeft
+        deptNamesText = map head $ map (T.splitOn (T.pack bracketLeftString)) convertToStr
+    in deptNamesText
+
+-- | Takes a Char and a String and returns whether or not that Char is in the String.
+containsChar :: Char -> String -> Bool
+containsChar _ [] = False
+containsChar y (x:xs)
+    | y == x = True
+    | otherwise = containsChar y xs  
+
+-- | Takes a Char and a String and checks whether that char is in the String. If it
+-- is, it returns the original string, otherwise it returns an empty string.
+filterList :: Char -> String -> String 
+filterList _ [] = ""
+filterList y (x:xs) 
+    | y == x = (t:ts)
+    | otherwise = filterList y xs  
+    where (t:ts) = (x:xs)
+
+-- | Removes the chars before and including the Char input in the given String.
```

Comment:
Ill reword the doc but for instance if we have the string "University" and the char 'v', the function will return "ersity". The way the Tag Texts were made it simple to break up the dept names from the dept codes with this function.